src/libData/DataStructures/TraceableDB.h

```diff
@@ -110,9 +116,22 @@ class TraceableDB : public dev::OverlayDB {
  public:
   void DetachedExecutePurge() {
     LOG_MARKER();
-    auto detached_func = [this]() { return ExecutePurge(0, true); };
+    auto detached_func = [this]() -> void {
+      if (!m_purgeRunning) {
+        m_purgeRunning = true;
```

Comment:
set m_stopSignal to false otherwise ExecutePurge might stop immediately

Reply:
It is false by default and also set false after every function, it won't be true if the thread isn't running.